modules/services/picom.nix

```diff
@@ -301,15 +299,13 @@ in {
 
       Service = let
         experimentalBackendsFlag =
-          if cfg.experimentalBackends then " --experimental-backends" else "";
+          lib.optionalString cfg.experimentalBackends "--experimental-backends";
       in {
-        ExecStart = "${cfg.package}/bin/picom --config ${configFile}"
-          + experimentalBackendsFlag;
+        ExecStart = "${cfg.package}/bin/picom --config ${
+            config.xdg.configHome + "/picom/picom.conf"
```

Comment:
I tried to use `xdg.configFile."picom/picom.conf".target` instead, however this generated `.config/picom/picom.conf` instead of `/home/hm-user/.config/picom/picom.conf`.

If there is a better solution I am all ears.

Reply:
There's `xdg.configFile.*.source` that points to the resulting file in the Nix store.

AFAIK there will only be a service restart if the settings are changed.

Reply:
> There's xdg.configFile.*.source that points to the resulting file in the Nix store.

If possible I would much prefer the way it currently is: pointing to the file inside the HOME. This way it is much easier to debug (e.g.: `systemctl cat --user picom.service` will show that the file is on `~/.config/picom/picom.conf`, so the user can open the file and see the contents).

> AFAIK there will only be a service restart if the settings are changed.

Not sure if I understood this comment here.

Reply:
I think in general it is best to point directly to the configuration file in the Nix store. I.e., `${config.xdg.configFile."picom/picom.conf".source}` in this case as @berbiche mentioned.

The `systemctl cat --user picom.service` should then contain the path to the configuration file in the Nix store so it should still be reasonably easy to find the file.

As mentioned we then also get the benefit that the systemd service will be automatically restarted (assuming `systemd.user.startServices` is enabled) when the configuration is changed. The tool that does the restarts essentially compares the old and new service files and does the restart if they differ.

Reply:
Btw, if you do switch to point directly to the configuration in the Nix store then you can still have stable tests using the `normalizeStorePaths` function. See

https://github.com/nix-community/home-manager/blob/7da4e6680f814562f070bad01922d6736e54be94/tests/modules/services/mpd/basic-configuration.nix#L11-L12

for an example

Reply:
Done.